### PR TITLE
Webui creates maintenance request without revision

### LINKED ISSUE #3546: webui maintenance_incident request creation happens without addrevision parameter
Issue/Feature description
====================

The webui is creating unversioned maintenance_incident requests unlike osc is doing. This breaks the review process (so I assume these requests will get refused in future).

The addrevision=1 parameter can be used to enabling the current revision, so the webui should use this as well.

Expected result
============

maintenance_incident actions should have a rev= attribute set by default.

How to Reproduce
===============
This got reported by Andreas Stieger:

Steps:
* make a normal maintenance branch, make a "good" change
* submit via web UI or without "addrevision"
* get the request reviewed
* make an "evil" change

### PR DESCRIPTION
**Pull request by eduardoj**
This should fix #3546.

Add missing `set_add_revision` call when creating a new incident request.

Add a check in the maintenance workflow for the source revision.

### CODE REVIEW

File: src/api/spec/features/webui/maintenance_workflow_spec.rb [hunk 1]
```diff
@@ -54,6 +54,17 @@
...
 
+    # Check that sending maintenance updates adds the source revision
+    new_bs_request_action = BsRequestAction.where(
+      type:                  "maintenance_incident",
+      target_project:        maintenance_project.name,
+      target_releaseproject: update_project.name,
+      source_project:        "#{user.home_project}:branches:#{update_project.name}",
+      source_package:        package.name
+    )
+    expect(new_bs_request_action).to exist
```
Comment (hennevogel) [inline src/api/spec/features/webui/maintenance_workflow_spec.rb]: *nitpick*: huh?

Comment (eduardoj) [inline src/api/spec/features/webui/maintenance_workflow_spec.rb]: Hm...  To be sure `new_bs_request_action` exists, before `source_rev` is checked. Hm... Should I drop this line?

Comment (bgeuken) [inline src/api/spec/features/webui/maintenance_workflow_spec.rb]: To give some background, we added this when we thought we need to call ``` bs_request.bs_request_actions.first.source_rev ```.

You can drop that line and change the following one to ``` expect(new_bs_request_action.pluck(:source_rev).first).not_to be nil ```

Comment (eduardoj) [inline src/api/spec/features/webui/maintenance_workflow_spec.rb]: @hennevogel @bgeuken : Changed to the line @bgeuken proposed.

**Pull request merged**